### PR TITLE
MM-47112 :  Migrate "components/interactive_dialog/dialog_introduction_text.jsx" to Typescript

### PR DESCRIPTION
**Pull request by VishakhaPoonia**
#### Summary
Migrate dialog introduction text component to typescript

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/21058
  Fixes https://mattermost.atlassian.net/browse/MM-47112

#### Related Pull Requests
NA

#### Screenshots
NA

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/interactive_dialog/dialog_introduction_text.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import * as Markdown from 'utils/markdown';
+import {getSiteURL} from 'utils/url';
+
```
Comment (M-ZubairAhmed) [inline components/interactive_dialog/dialog_introduction_text.tsx:8]: nit : no empty line

**Review by M-ZubairAhmed — APPROVED** [review]

File: components/interactive_dialog/dialog_introduction_text.test.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import {mount} from 'enzyme';
 
 import EmojiMap from 'utils/emoji_map';
 
-import DialogIntroductionText from './dialog_introduction_text.jsx';
+import DialogIntroductionText from './dialog_introduction_text.tsx';
```
Comment (nevyangelova) [inline components/interactive_dialog/dialog_introduction_text.test.jsx:10]: Thanks @VishakhaPoonia would it be possible to migrate the test to typescript as well? Just by passing the prop types to the test as well.

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
As @nevyangelova suggested lets also migrate the file's test to ts as well. And few nits

File: components/interactive_dialog/dialog_introduction_text.tsx (hunk 1 above)
Comment (VishakhaPoonia) [reply components/interactive_dialog/dialog_introduction_text.tsx:8]: @M-ZubairAhmed To be fair these empty line changes are a bit annoying, please add lint rules as soon as possible

**Review by M-ZubairAhmed — COMMENTED** [review]
🆗  It looks good to me. Thank you

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by nevyangelova — APPROVED** [review]

**Review by AshishDhama — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
@AshishDhama should we have QA on this one. since it has one component file. Your call!

**Comment by AshishDhama** [comment]
@M-ZubairAhmed if e2e are passing I think we can Defer QA.

**Review by jgilliam17 — APPROVED** [review]
Thanks @VishakhaPoonia 
Tested, looks good to merge.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/248343-e8c9331-onprem-ent-webapp-pr-11169-mm-ee-test:e8c9331/mochawesome.html) - no PR related failures.

**Pull request merged**